File: Languages/Julia/web_server.jl
# Hello, World!
println("Hello, World!")

using HTTP
using JSON

function make_http_request(url)
    try
        response = HTTP.get(url; timeout=5, readtimeout=5)
        if response.status == 200
            return String(response.body)
        else
            return ""
        end
    catch e
        return ""
    end
end

function test_web_server()
    base_url = "http://localhost:3000"
    compare_file = "../webserver/compare.json"
    output_file = "test-result.json"
    
    # Load expected hashes
    if !isfile(compare_file)
        println("Error: Could not find $compare_file")
        return
    end
    
    expected_hashes = JSON.parsefile(compare_file)
    
    println("Testing 100 endpoints...")
    
    results = []
    passed = 0
    failed = 0
    
    # Test each endpoint
    for i in 0:99
        endpoint = "test-$i"
        url = "$base_url/$endpoint"
        
        response = make_http_request(url)
        
        result = Dict(
            "endpoint" => endpoint,
            "url" => url,
            "expected_hash" => get(expected_hashes, endpoint, "")
        )
        
        if response == ""
            result["status"] = "FAILED"
            result["error"] = "HTTP request failed"
            failed += 1
        else
            try
                data = JSON.parse(response)
                server_hash = get(data, "hash", "")
                result["server_hash"] = server_hash
                
                if server_hash == result["expected_hash"]
                    result["status"] = "PASSED"
                    passed += 1
                else
                    result["status"] = "FAILED"
                    failed += 1
                end
            catch e
                result["status"] = "FAILED"
                result["error"] = "JSON parse error"
                failed += 1
            end
        end
        
        push!(results, result)
        
        # Progress indicator
        if (i + 1) % 10 == 0
            println("Tested $(i + 1)/100 endpoints...")
        end
    end
    
    # Create final result
    total_tests = length(results)
    success_rate = round(passed / total_tests * 100, digits=1)
    
    final_result = Dict(
        "total_tests" => total_tests,
        "passed" => passed,
        "failed" => failed,
        "success_rate" => "$(success_rate)%",
        "timestamp" => string(now()),
        "results" => results
    )
    
    # Save results to file
    try
        open(output_file, "w") do f
            JSON.print(f, final_result, 2)
        end
        println("Results saved to: $output_file")
    catch e
        println("Error saving results: $e")
    end
    
    println("\nTest completed!")
    println("Passed: $passed/$total_tests ($(success_rate)%)")
    println("Failed: $failed/$total_tests ($(100 - success_rate)%)")
end

test_web_server()

File: Languages/Ruby/web_server.rb
# Hello, World!
puts "Hello, World!"

require 'net/http'
require 'json'
require 'uri'

def make_http_request(url)
  uri = URI(url)
  http = Net::HTTP.new(uri.host, uri.port)
  http.read_timeout = 5
  http.open_timeout = 5
  
  request = Net::HTTP::Get.new(uri)
  response = http.request(request)
  
  if response.code == '200'
    response.body
  else
    ''
  end
rescue
  ''
end

def test_web_server
  base_url = "http://localhost:3000"
  compare_file = "../webserver/compare.json"
  output_file = "test-result.json"
  
  # Load expected hashes
  begin
    expected_hashes = JSON.parse(File.read(compare_file))
  rescue => e
    puts "Error: Could not read #{compare_file}: #{e.message}"
    return
  end
  
  puts "Testing 100 endpoints..."
  
  results = []
  passed = 0
  failed = 0
  
  # Test each endpoint
  (0..99).each do |i|
    endpoint = "test-#{i}"
    url = "#{base_url}/#{endpoint}"
    
    response = make_http_request(url)
    
    result = {
      endpoint: endpoint,
      url: url,
      expected_hash: expected_hashes[endpoint] || ''
    }
    
    if response.empty?
      result[:status] = "FAILED"
      result[:error] = "HTTP request failed"
      failed += 1
    else
      begin
        data = JSON.parse(response)
        server_hash = data['hash'] || ''
        result[:server_hash] = server_hash
        
        if server_hash == result[:expected_hash]
          result[:status] = "PASSED"
          passed += 1
        else
          result[:status] = "FAILED"
          failed += 1
        end
      rescue JSON::ParserError
        result[:status] = "FAILED"
        result[:error] = "JSON parse error"
        failed += 1
      end
    end
    
    results << result
    
    # Progress indicator
    if (i + 1) % 10 == 0
      puts "Tested #{i + 1}/100 endpoints..."
    end
  end
  
  # Create final result
  final_result = {
    total_tests: results.length,
    passed: passed,
    failed: failed,
    success_rate: "#{(passed.to_f / results.length * 100).round(1)}%",
    timestamp: Time.now.strftime("%Y-%m-%d %H:%M:%S"),
    results: results
  }
  
  # Save results to file
  begin
    File.write(output_file, JSON.pretty_generate(final_result))
    puts "Results saved to: #{output_file}"
  rescue => e
    puts "Error saving results: #{e.message}"
  end
  
  puts "\nTest completed!"
  puts "Passed: #{passed}/#{results.length} (#{(passed.to_f / results.length * 100).round(1)}%)"
  puts "Failed: #{failed}/#{results.length} (#{(failed.to_f / results.length * 100).round(1)}%)"
end

test_web_server
